gapic/ads-templates/%namespace/%name/%version/%sub/services/%service/client.py.j2

```diff
@@ -381,6 +382,14 @@ class {{ service.client_name }}(metaclass={{ service.client_name }}Meta):
             {%- endfor %} {# key, field in method.flattened_fields.items() #}
             {%- endif %} {# method.client_streaming #}
 
+        {%- if method.name|snake_case == "pull" %}
+        if request.return_immediately:
+            warnings.warn(
+                "The return_immediately flag is deprecated and should be set to False.",
+                category=DeprecationWarning,
+            )
+        {%- endif %}
```

Comment:
I am in general not comfortable adding this change. It's extremely specific to one method of one API, the check is too generic for safety, and it complicates the template for a warning. People mostly ignore warnings.

Reply:
+1 to this seeming too generic to keep here. If some other API happens to have a field with the same name it will result in a spurious warning.

It looks like this was field was marked deprecated in the protos last year.

https://github.com/googleapis/googleapis/blob/a1ab4d44db02d59ff58810c6d4182d84e4b9abaa/google/pubsub/v1/pubsub.proto#L994-L1003
https://github.com/googleapis/googleapis/commit/01c681586d8d6dbd60155289b587aee678530bd9

I think ideally the generators would notice fields marked 'deprecated' and automatically update the docstring and raise a warning.

Reply:
What would be a reasonable alternative in your opinion? We can still hack it in through `synth.py`, but the regex will likely be quite messy.

There's also an ongoing initiative (AFAIK?) to move most of the code from `synth.py` into the generator itself (please correct me if I'm wrong on that).

Reply:
I think this is primarily a documentation problem. That documentation can take multiple concurrent forms: docstrings, samples with comments, and hand written documention. I am tentatively in favor of the generator checking for the `deprecated` flag and autogenerating some docstring boilerplate and mostly **not** in favor of printing warnings.

Reply:
I think that there are likely to always be situations where the most reasonable and practical solution is `synth.py` hacks. Removing most of the code from `synth.py` is definitely an explicit goal, but moving oneoff fixes into the generator are half solutions.

Reply:
Re: deprecation warning - here it's an explicit [request](https://github.com/googleapis/python-pubsub/issues/350), was decided in one of the meetings. I was not present, though, thus I'll re-check if it is a hard requirement.